docs/content/concepts/assets/software-defined-assets.mdx

```diff
@@ -0,0 +1,341 @@
+---
+title: Software-Defined Assets | Dagster
+description: A software-defined asset is a description of how to compute the contents of a particular data asset.
+---
+
+# Software-Defined Assets
+
+A software-defined asset is a description of how to compute the contents of a particular data asset.
+
+## Relevant APIs
+
+| Name                                  | Description                                                            |
+| ------------------------------------- | ---------------------------------------------------------------------- |
+| <PyObject object="asset" decorator /> | A decorator used to define assets.                                     |
+| <PyObject object="SourceAsset" />     | A class that describes an asset, but doesn't define how to compute it. |
```

Comment:
This feels a little unclear to me what a source asset is. Can we say something instead like "An asset whose computation is defined externally from the repository"?

Reply:
For many source assets, their computation isn't really defined anywhere. E.g. a source asset might represent a file that someone manually uploads to a server every so often. I'm definitely open to workshopping this to make it more clear though. Do you have sense of the source of the lack of clarity?

Reply:
I see. To me, it feels unclear because I would want to read the source asset description and know what I would use it for, and with this description I'm not sure. Some options:
- An asset that is not generated by a Dagster op
- An asset that is defined externally from the repository